File: internal/core/bus.go
package core

type EventBus struct {
	ch chan Event
}

func NewEventBus(buffer int) *EventBus {
	return &EventBus{ch: make(chan Event, buffer)}
}

func (b *EventBus) Publish(event Event) {
	b.ch <- event
}

func (b *EventBus) Subscribe() <-chan Event {
	return b.ch
}

func (b *EventBus) QSize() int {
	return len(b.ch)
}

File: pidex/core/bus.py
import queue
from pidex.core.event import Event


class EventBus:
    def __init__(self):
        self._queue: queue.Queue = queue.Queue()

    def publish(self, event: Event) -> None:
        self._queue.put_nowait(event)

    def get(self, timeout: float | None = None) -> Event:
        return self._queue.get(timeout=timeout)

    @property
    def qsize(self) -> int:
        return self._queue.qsize()
